extensions-contrib/druid-jose4j/src/main/java/org/apache/druid/security/jose4j/Jose4jAuthenticator.java

```diff
@@ -0,0 +1,197 @@
+    this.issuer = Preconditions.checkNotNull(issuer, "null issuer");
+    this.jwksUri = Preconditions.checkNotNull(jwksUri, "null jwksUri");
+    this.authorizerName = authorizerName;
+    this.audience = audience;
+    this.skipOnFailure = skipOnFailure == null ? false : skipOnFailure;
+    this.jwtAuthenticator = new JwtAuthenticator(issuer, jwksUri, audience);
+  }
+
+  @Override
+  public Filter getFilter()
+  {
+    return new HTTPAuthenticationFilter();
+  }
+
+  @Override
+  public Class<? extends Filter> getFilterClass()
+  {
+    return HTTPAuthenticationFilter.class;
+  }
+
+  @Override
+  public String getAuthChallengeHeader()
+  {
+    return "Bearer";
+  }
+
+  @Override
+  public Map<String, String> getInitParameters()
+  {
+    return null;
+  }
+
+  @Override
+  public String getPath()
+  {
+    return "/*";
+  }
+
+  @Override
+  public EnumSet<DispatcherType> getDispatcherType()
+  {
+    return null;
+  }
+
+
+  @Override
+  @Nullable
+  public AuthenticationResult authenticateJDBCContext(Map<String, Object> context)
+  {
+    String bearerToken = (String) context.get("token");
+
+    if (bearerToken != null) {
+      try {
+        JwtClaims jwtClaims = jwtAuthenticator.authenticate(bearerToken);
+        if (jwtClaims != null) {
+          LOG.info("JWT validation succeeded!");
+          return new AuthenticationResult(
+              jwtClaims.getSubject(),
+              authorizerName,
+              name,
+              null
+          );
+        }
+      }
+      catch (InvalidJwtException | MalformedClaimException ex) {
+        LOG.error(ex, "Invalid Credentails");
+      }
+    }
+    return null;
+  }
+
+  public class HTTPAuthenticationFilter implements Filter
+  {
+    @Override
+    public void init(FilterConfig filterConfig)
+    {
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+        throws IOException, ServletException
+    {
+      HttpServletResponse httpResp = (HttpServletResponse) servletResponse;
+      String bearerToken = AuthUtils.getBearerTokenFromHttpReq((HttpServletRequest) servletRequest);
+      if (bearerToken == null) {
+        // Request didn't have HTTP bearer token credentials, move on to the next filter
+        filterChain.doFilter(servletRequest, servletResponse);
+        return;
+      }
+      try {
+        JwtClaims jwtClaims = jwtAuthenticator.authenticate(bearerToken);
+        if (jwtClaims != null) {
+          LOG.info("JWT validation succeeded!");
+          AuthenticationResult authenticationResult = new AuthenticationResult(
+              jwtClaims.getSubject(),
+              authorizerName,
+              name,
+              null
+          );
+          servletRequest.setAttribute(AuthConfig.DRUID_AUTHENTICATION_RESULT, authenticationResult);
+          filterChain.doFilter(servletRequest, servletResponse);
+        } else {
+          if (skipOnFailure) {
```

Comment:
is this a standard flag for other authenticators or did we just add this for testing?

Reply:
it's a standard flag for authenticators. It's used primarily on more than 2 instances of the same type of authenticators.
In the revised version, I removed this flag and added an issuer extractor to filter the JWT verifier by issuer